### PR TITLE
Strings / it-academy.by / JC1-110-22 / Maksim Bondar / Tasks 1 - 14 complete

### PR DESCRIPTION
**Pull request by shiv-qa-auto**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+public class StringUtilTest {
+    StringUtil stringUtil = new StringUtil();
+
+    //Task 1 tests
+    @Test
+    void concatTwoStringsTest() {
+        String expected = "789123";
+        String actual = stringUtil.concatTwoStrings("789", "123");
+        Assertions.assertEquals(actual, expected);
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java:14]: Use approach GIVEN empty line WHEN empty line THAN
```suggestion
        String expected = "789123";
        
        String actual = stringUtil.concatTwoStrings("789", "123");
        
        Assertions.assertEquals(actual, expected);
```

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
+package com.rakovets.course.java.core.practice.strings;
+
+public class StringUtil {
+
+    //Task 1
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    public String concatTwoStrings(String one, String two) {
+        return one + two;
+    }
+
+    //Task 2
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    public int findStringSymbolIndex(String one, char x) {
+        int index = 0;
+        if (index < one.length()) {
+            index = one.indexOf(x);
+        } else {
+            index = -1;
+        }
+        return index;
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    //Task 3
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    public boolean isStringsEquals(String one, String two) {
+        return one.equals(two);
+    }
+
+    //Task 4
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    public String stringTrimAndUpperCase(String one) {
+        return one.trim().toUpperCase();
+    }
+
+    //Task 5
+    public String removeSubstring(String one, int fromIndex, int toIndex) {
+        return one.substring(fromIndex, toIndex);
+    }
+
+    //Task 6
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    public String replaceSadFaceInString(String one) {
+        return one.replace(":(", ":)");
+    }
+
+    //Task 7
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        for (char c : one.toCharArray()) {
+            c = Character.toLowerCase(c);
+            if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || c == 'y') {
+                vowelsCounter++;
+            }
+        }
+        return vowelsCounter;
+    }
+
+    //Task 9
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+        if (one.length() > 0) {
+            String lowerCase = one.toLowerCase();
+            StringBuilder stringBuffer = new StringBuilder(lowerCase);
+            stringBuffer.reverse();
+            result = lowerCase.equals(stringBuffer.toString());
+        } else result = false;
+        return result;
+    }
+
+    //Task 11
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+            if (i == arrayLength - 1) {
+                array[i] = one.substring(i * symbolsPerString);
+            } else {
+                array[i] = one.substring(i * symbolsPerString, i * symbolsPerString + symbolsPerString);
+            }
+        }
+        return array;
+    }
+
+    //Task 12
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+        return wordsCounter;
+    }
+
+    //Task 13
+    public String parseInitialsFromNameAndSurname(String one) {
+        String[] arrayOfInitials = one.trim().toUpperCase().split(" ");
+        return arrayOfInitials[0].charAt(0) + arrayOfInitials[1].substring(0,1);
+    }
+
+    //Task 14
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+    public String parseDigitsFromString(String one) {
+        return one.replaceAll("\\D", "");
+    }
+
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/strings/StringUtil.java:103]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 14]
```diff
@@ -0,0 +1,335 @@
+package com.rakovets.course.java.core.practice.strings;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class StringUtilTest {
+    StringUtil stringUtil = new StringUtil();
+
+    //Task 1 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 15]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void isStringsEqualsBothEmptyTest() {
+        boolean expected = true;
+        boolean actual = stringUtil.isStringsEquals("", "");
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 4 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 16]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void stringTrimAndUpperCaseAlreadyUpperCaseTest() {
+        String expected = "TEST";
+        String actual = stringUtil.stringTrimAndUpperCase("TEST");
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 5 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 17]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void removeSubstringSpacesTest() {
+        String expected = "test    ";
+        String actual = stringUtil.removeSubstring("This is test       ", 8, 16);
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 6 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 18]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void replaceSadFaceInStringMultipleEmojiTest() {
+        String expected = ":):):):)";
+        String actual = stringUtil.replaceSadFaceInString(":(:(:(:(");
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 7 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 19]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void isStringStartAndFinishWithEmptyStringAndTestTest() {
+        boolean expected = true;
+        boolean actual = stringUtil.isStringStartAndFinishWith(" ", " ");
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 8 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 20]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void vowelsCounterInStringUpperAndLowerCaseTest() {
+        int expected = 11;
+        int actual = stringUtil.vowelsCounterInString("thIs is test messagE for the cOunter");
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 9 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 21]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void specialSymbolsCounterInStringNoSymbolsTest() {
+        int expected = 0;
+        int actual = stringUtil.specialSymbolsCounterInString("My name is test");
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 10 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 22]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void isStringPalindromeEmptyStringTest() {
+        boolean expected = false;
+        boolean actual = stringUtil.isStringPalindrome("");
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 11 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 23]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void stringSplitToArraysAllSymbolsInOneStringTest() {
+        String[] expected = {"Hello"};
+        String[] actual = stringUtil.stringSplitToArrays("Hello", 5);
+        Assertions.assertArrayEquals(actual, expected);
+    }
+
+    //Task 12 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 24]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void wordsInStringCounterMultipleSpacesTest() {
+        int expected = 6;
+        int actual = stringUtil.wordsInStringCounter("Hey   there!  This is    test message.");
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 13 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java [hunk 25]
```diff
@@ -0,0 +1,335 @@
...
+    }
+
+    @Test
+    void parseInitialsFromNameAndSurnameLowerCaseTest() {
+        String expected = "MB";
+        String actual = stringUtil.parseInitialsFromNameAndSurname("maksim bondar");
+        Assertions.assertEquals(actual, expected);
+    }
+
+    //Task 14 tests
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

File: src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java (hunk 1 above)
Comment (shiv-qa-auto) [reply src/test/java/com/rakovets/course/java/core/practice/strings/StringUtilTest.java]: Fixed, please review

**Review by rakovets — APPROVED** [review]

**Pull request closed without merging**